### PR TITLE
Adding data service api layer

### PR DESCRIPTION
**Pull request by narehayrapetyan**
Adding data service api layer

### CODE REVIEW

**Comment by wjbuys** [comment]
This one should be rebased against current `dev` branch before review, right?

File: data-service-model/src/main/java/com/amazonaws/blox/dataservicemodel/v1/model/wrappers/DescribeEnvironmentResponse.java [hunk 1]
```diff
@@ -41,8 +39,8 @@
   @NonNull private InstanceGroup instanceGroup;
   @NonNull private EnvironmentType type;
   @NonNull private String environmentVersion;
-  @NonNull private EnvironmentStatus status;
-  @NonNull private EnvironmentHealth health;
+  @NonNull private String status;
```
Comment (wjbuys) [inline data-service-model/src/main/java/com/amazonaws/blox/dataservicemodel/v1/model/wrappers/DescribeEnvironmentResponse.java:42]: Why the change from enum to string?

**Review by wjbuys — APPROVED** [review]

File: data-service-model/src/main/java/com/amazonaws/blox/dataservicemodel/v1/model/wrappers/DescribeEnvironmentResponse.java (hunk 1 above)
Comment (narehayrapetyan) [reply data-service-model/src/main/java/com/amazonaws/blox/dataservicemodel/v1/model/wrappers/DescribeEnvironmentResponse.java:42]: Because when you have enums on the server and client side making changes becomes very difficult as it's hard to guarantee that everyone has the latest changes at the same time. Constants allow us to have access to allowed values while avoiding the deserialization problem with enums.
